lib/launcher.rb

```diff
@@ -119,18 +119,18 @@ def webrick_server(app)
     end
 
     def launch
-      ::Proxy::PluginInitializer.new(::Proxy::Plugins.instance).initialize_plugins
-
-      http_app = http_app()
-      https_app = https_app()
-      raise Exception.new("Both http and https are disabled, unable to start.") if http_app.nil? && https_app.nil?
-
       if SETTINGS.daemon
         check_pid
         Process.daemon
         write_pid
       end
 
+      ::Proxy::PluginInitializer.new(::Proxy::Plugins.instance).initialize_plugins
+
+      http_app = http_app()
+      https_app = https_app()
+      raise Exception.new("Both http and https are disabled, unable to start.") if http_app.nil? && https_app.nil?
```

Comment:
Hmm. I think this is going to be tricky -- here we effectively check that there is at least one module successfully configured using the appropriate port. Ports used (:https_enabled and :http_enabled properties in elements of ::Proxy::Plugins hash) themselves are set during plugin initialization.